### PR TITLE
MM-27229 - Cypress/E2E: Automate backlogs - BOT Accounts - Part 06 (10 test cases)

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
automated:
MM-T1861
MM-T1862
MM-T1863
MM-T1864
MM-T1865
MM-T1866
MM-T1867
MM-T1868
MM-T1869
MM-T1870

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-27229
#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

### CODE REVIEW

**Review by calebroseland — COMMENTED** [review]
There are some errors on local execution of `managing_bot_accounts_spec.js`. Is `cypress/fixtures/com.github.matterpoll.matterpoll.tar.gz` missing?

**Comment by reflog** [comment]
@calebroseland - this and some other tests require downloading files . see `poll_spec.js` line 10

**Review by calebroseland — APPROVED** [review]

**Review by prapti — APPROVED** [review]
Test passed in subsequent runs (might need to look into this again but they're passing now).
Steps and changes look good to me!
Approving.

File: e2e/cypress/integration/bot_accounts/bot_api_spec.js [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+            const msg1 = 'this is a bot message ' + botName;
+            const msg2 = 'this is a bot message2 ' + botName;
+            cy.apiCreatePost(newChannel.id, msg1, '', {attachments: [{pretext: 'Look some text', text: 'This is text'}]}, token).then(({body: post1}) => {
+                cy.apiCreatePost(newChannel.id, msg2, '', {attachments: [{pretext: 'Look some text', text: 'This is text'}]}, token).then(({body: post2}) => {
+                    // # Re-login to validate post presence
+                    cy.apiAdminLogin();
+                    cy.visit(`/${newTeam.name}/channels/` + newChannel.name);
+
+                    // * Validate posts were created
+                    cy.get(`#postMessageText_${post1.id}`).should('contain', msg1);
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/bot_api_spec.js:194]:
```suggestion
                    cy.get(`#postMessageText_${post1.id}`, {timeout: TIMEOUTS.ONE_MIN}).should('contain', msg1);
```

File: e2e/cypress/integration/bot_accounts/bot_api_spec.js [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+                    cy.get(`#postMessageText_${post2.id}`).should('contain', msg2);
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/bot_api_spec.js:195]:
```suggestion
                    cy.get(`#postMessageText_${post2.id}`, {timeout: TIMEOUTS.ONE_MIN}).should('contain', msg2);
```

File: e2e/cypress/integration/bot_accounts/bot_api_spec.js [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+            const msg1 = 'this is a bot message ' + botName;
+            cy.apiCreatePost(newChannel.id, msg1 + ' to @sysadmin', '', {}, token);
+
+            // # Re-login to validate post presence
+            cy.apiAdminLogin();
+            cy.visit(`/${newTeam.name}/channels/` + newChannel.name);
+
+            cy.getLastPostId().then((postId) => {
+                // * Validate post was created
+                cy.get(`#postMessageText_${postId}`).should('contain', msg1);
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/bot_api_spec.js:224]:
```suggestion
                cy.get(`#postMessageText_${postId}`, {timeout: TIMEOUTS.ONE_MIN}).should('contain', msg1);
```

File: e2e/cypress/integration/bot_accounts/bot_api_spec.js [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+                // * Assert that the last message posted contains highlighted mention
+                cy.get(`#postMessageText_${postId}`).find('.mention--highlight').should('be.visible');
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/bot_api_spec.js:227]:
```suggestion
                cy.get(`#postMessageText_${postId}`, {timeout: TIMEOUTS.ONE_MIN}).find('.mention--highlight').should('be.visible');
```

File: e2e/cypress/integration/bot_accounts/bot_api_spec.js [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+                // # Create a post
+                cy.apiCreatePost(channel.id, msg1 + ' to @sysadmin', '', {}, token);
+
+                // # Re-login to validate post presence
+                cy.apiAdminLogin();
+                cy.visit(`/${newTeam.name}/channels/` + channel.name);
+
+                cy.getLastPostId().then((postId) => {
+                    // * Validate post was created
+                    cy.get(`#postMessageText_${postId}`).should('contain', msg1);
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/bot_api_spec.js:294]:
```suggestion
                    cy.get(`#postMessageText_${postId}`, {timeout: TIMEOUTS.ONE_MIN}).should('contain', msg1);
```

**Review by prapti — CHANGES_REQUESTED** [review]
Actually, after running the tests again because they were failing intermittently, it looks like adding the timeout will take care of those failures.

**Comment by reflog** [comment]
thanks @prapti !

**Review by prapti — APPROVED** [review]
Thanks Eli! LGTM!

**Comment by prapti** [comment]
Took a quick look at why lint is complaining here and looks like the following may need to be addressed as well:
`/root/mattermost/mattermost-webapp/e2e/cypress/integration/bot_accounts/managing_bot_accounts_spec.js
  287:10  error  'createCustomAdmin' is defined but never used  no-unused-vars
  288:26  error  'generateRandomUser' is not defined            no-undef`

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- All tests return green after deploying the latest version
- Ensured that TM4J tests are accurately captured by these Cy tests
- No issues found 
LGTM!

All corresponding TM4J task have Cy status correctly set to `update`.

**Pull request merged**